src/main/scala/com/databricks/labs/mosaic/core/Mosaic.scala

```diff
@@ -82,6 +82,14 @@ object Mosaic {
                     val kRing = indexSystem.kRing(current, 1)
                     val toQueue = kRing.filterNot(newTraversed.contains)
                     (toQueue, accumulator._2 ++ Seq(chip))
+                } else if (newTraversed.size == 1) {
```

Comment:
it took me a second to realise this is "else if".
For as second I thought we would do kring for any first point.
This is the correct fix!

Reply:
The other approach would be to use st_touches between cell boundary and point, but due to numerical stuff it may still miss the start.
Another thing that would work well, would be to change lines 64 and 65 to start from all vertices simultaneously. 
It would still fail if the line is going along the index edge so KRing is the only thing that would abstract from that.

Reply:
Do you mean indexing all the points of the line before walking the k-ring? I considered that as well, but that needs some benchmarking to understand if in real-world scenarios that would be more or less efficient.

Reply:
yes, that is what I meant, in general it wont change much in the grand scheme of things.
It is a rare event to hit starting points on the grid border, so general benchmarks wont be affected